lib/experimental/fonts-library/class-wp-font-family.php

```diff
@@ -0,0 +1,421 @@
+			if ( $new_src ) {
+				$new_font_face['src'][] = $new_src;
+			}
+		}
+		if ( count( $new_font_face['src'] ) === 1 ) {
+			$new_font_face['src'] = $new_font_face['src'][0];
+		}
+		return $new_font_face;
+	}
+
+
+	/**
+	 * Downloads font face assets if the font family is a Google font, or moves them if it is a local font.
+	 *
+	 * @param array $files An array of files to be installed.
+	 * @return bool|WP_Error
+	 */
+	private function download_or_move_font_faces( $files ) {
+		if ( $this->has_font_faces() ) {
+			$new_font_faces = array();
+			foreach ( $this->data['fontFace'] as $font_face ) {
+				if ( empty( $files ) ) {
+					// If we are installing local fonts, we need to move the font face assets from the temp folder to the wp fonts directory.
+					$new_font_face = $this->download_font_face_assets( $font_face );
+				} else {
+					// If we are installing google fonts, we need to download the font face assets.
+					$new_font_face = $this->move_font_face_asset( $font_face, $files[ $font_face['file'] ] );
+				}
+				// If the font face assets were successfully downloaded, we add the font face to the new font.
+				// Font faces with failed downloads are not added to the new font.
+				if ( ! empty( $new_font_face['src'] ) ) {
+					$new_font_faces[] = $new_font_face;
+				}
+			}
+			if ( ! empty( $new_font_faces ) ) {
+				$this->data['fontFace'] = $new_font_faces;
+				return true;
+			}
+			return false;
+		}
+		return true;
+	}
+
+	/**
+	 * Get the post for a font family.
+	 *
+	 * @return WP_Post|null The post for this font family object or null if the post does not exist.
+	 */
+	public function get_font_post() {
+		$args = array(
+			'post_type'      => 'wp_fonts_library',
+			'post_name'      => $this->data['slug'],
+			'name'           => $this->data['slug'],
+			'posts_per_page' => 1,
+		);
+
+		$posts_query = new WP_Query( $args );
+
+		if ( $posts_query->have_posts() ) {
+			$post = $posts_query->posts[0];
+			return $post;
+		}
+
+		return null;
+	}
+
+	/**
+	 * Get the data for this object from the database and set it to the data property.
+	 *
+	 * @return WP_Post|null The post for this font family object or null if the post does not exist.
+	 */
+	private function get_data_from_post() {
+		$post = $this->get_font_post();
+		if ( $post ) {
+			$data       = json_decode( $post->post_content, true );
+			$this->data = $data;
+			return $post;
+		}
+		return null;
+	}
+
+	/**
+	 * Create a post for a font family.
+	 *
+	 * @return int post id
+	 */
+	private function create_font_post() {
+		$post    = array(
+			'post_title'   => $this->data['name'],
+			'post_name'    => $this->data['slug'],
+			'post_type'    => 'wp_fonts_library',
+			'post_content' => $this->get_data_as_json(),
+			'post_status'  => 'publish',
+		);
+		$post_id = wp_insert_post( $post );
+		if ( 0 === $post_id ) {
+			return WP_Error( 'font_post_creation_failed', __( 'Font post creation failed', 'gutenberg' ) );
```

Comment:
Possibly, a fatal error:
```suggestion
			return new WP_Error( 'font_post_creation_failed', __( 'Font post creation failed', 'gutenberg' ) );
```